Yank/analyze.py

```diff
@@ -535,13 +555,28 @@ def generate_mixing_statistics(self, number_equilibrated=0):
             Eigenvalues of the Transition matrix sorted in descending order
         """
 
-        # Get mixing stats from reporter
-        n_accepted_matrix, n_proposed_matrix = self._reporter.read_mixing_statistics()
-        # Add along iteration dim
-        n_accepted_matrix = n_accepted_matrix[number_equilibrated:].sum(axis=0).astype(float)  # Ensure float division
-        n_proposed_matrix = n_proposed_matrix[number_equilibrated:].sum(axis=0)
-        # Compute empirical transition count matrix
-        t_ij = 1 - n_accepted_matrix/n_proposed_matrix
+        # Read data from disk
+        states = self._reporter.read_replica_thermodynamic_states()
+        n_iterations, n_states = states.shape
+        n_ij = np.zeros([n_states, n_states], np.int64)
+
+        # Compute empirical transition count matrix.
+        for iteration in range(number_equilibrated, n_iterations - 1):
+            for i_replica in range(n_states):
+                i_state = states[iteration, i_replica]
+                j_state = states[iteration + 1, i_replica]
+                n_ij[i_state, j_state] += 1
+
+        # Compute transition matrix estimate.
+        # TODO: Replace with maximum likelihood reversible count estimator from msmbuilder or pyemma.
+        t_ij = np.zeros([n_states, n_states], np.float64)
+        for i_state in range(n_states):
+            denominator = (n_ij[i_state, :].sum() + n_ij[:, i_state].sum())
+            if denominator > 0:
+                for j_state in range(n_states):
```

Comment:
An optimization here could be to run only from `i_state + 1` to `n_states` and use symmetry to fill the other half of the matrix, but this can be done later.

Are we computing this from scratch to maintain backwards compatibility with calculations run before you implemented the fix in `mix_replicas`?